### PR TITLE
Add accessibility labels for notifications icon and filters

### PR DESCRIPTION
**Pull request by mazevedofs**
**Phabricator:** Partially addresses https://phabricator.wikimedia.org/T288776

### Notes
* This PR adds some accessibility labels to the notifications icon and the filters on the Notifications Center, to provide better context to VoiceOver users.
* The strings used here are also open to discussion, suggestions to improve the wording and context are appreciated

### Test Steps
1. Enable VoiceOver on the device - you can also use the accessibility inspector on Xcode
2. Tap the notifications icon on the Explore feed, both with and without the badge. Make sure that it correctly informs the user when the icon has the badge
3. On the Notifications Central, check if the user is correctly informed about the filters' states correctly, with and without filters engaged.

### CODE REVIEW

File: Wikipedia/Code/WMFAppViewController.m [hunk 1]
```diff
@@ -1417,7 +1417,7 @@ - (void)setNotificationsCenterButtonForExploreViewController:(ExploreViewControl
         NSInteger numUnreadNotifications = [[self.dataStore.remoteNotificationsController numberOfUnreadNotificationsAndReturnError:nil] integerValue];
         UIImage *image = numUnreadNotifications == 0 ? [self notificationsCenterBellImageWithUnreadNotifications:NO] : [self notificationsCenterBellImageWithUnreadNotifications:YES];
         UIBarButtonItem *notificationsBarButton = [[UIBarButtonItem alloc] initWithImage:image style:UIBarButtonItemStylePlain target:_exploreViewController action:@selector(userDidTapNotificationsCenter)];
-        notificationsBarButton.accessibilityLabel = [WMFCommonStrings notificationsCenterTitle];
+        notificationsBarButton.accessibilityLabel = numUnreadNotifications == 0 ? [WMFCommonStrings notificationsCenterTitle] : [WMFCommonStrings notificationsCenterBadgeTitle];
```
Comment (tonisevener) [inline Wikipedia/Code/WMFAppViewController.m:1420]: We'll need this same logic in `WMFSettingsViewController.m` too - currently it doesn't switch to the `notificationsCenterBadgeTitle` type when it shows the badged icon.

File: Wikipedia/Code/NotificationsCenterViewModel.swift [hunk 2]
```diff
@@ -315,6 +315,18 @@ extension NotificationsCenterViewModel {
     var typeFilterButtonImage: UIImage? {
         return toolbarImageForTypeFilter(engaged: remoteNotificationsController.filterState.offTypes.count > 0 || remoteNotificationsController.filterState.readStatus != .all)
     }
+    
+    var filterButtonAccessibilityLabel: String {
+        return remoteNotificationsController.filterState.offTypes.count > 0 ?
+        WMFLocalizedString("applied-filters-accesibility-label", value: "Notifications Filter - has filters applied", comment: "Notifications filters button is selected indicating that filters are enabled")
```
Comment (tonisevener) [inline Wikipedia/Code/NotificationsCenterViewModel.swift:321]: Go ahead and mention the accessibility piece in the comment of these strings:
```
WMFLocalizedString("notifications-center-applied-filters-accessibility-label", value: "Notifications Filter - has filters applied", comment: "Accessibility label for Notifications Center's filters button. This button is in a selected state indicating that filters are applied.")`

WMFLocalizedString("notifications-center-filters-accessibility-label", value: "Notifications Filter", comment: "Accessibility label for Notifications Center's filters button. This button is in an unselected state indicating that filters are not applied.")
```

File: Wikipedia/Code/WMFAppViewController.m (hunk 1 above)
Comment (mazevedofs) [reply Wikipedia/Code/WMFAppViewController.m:1420]: Great catch! I forgot to test with the explore feed off

**Review by tonisevener — APPROVED** [review]
Looks good! I pushed one minor commit for spelling and using your new `areFiltersApplied` computed property in the filter button image logic.

**Pull request merged**